### PR TITLE
feat: add short description to recipes

### PR DESCRIPTION
**Pull request by philasmar**
*Issue #, if available:*

*Description of changes:*
feat: add short description to recipes

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs [hunk 1]
```diff
@@ -121,7 +121,7 @@ public async Task<IActionResult> GetRecommendations(string sessionId)
                 output.Recommendations.Add(new RecommendationSummary(
                     recommendation.Recipe.Id,
                     recommendation.Name,
-                    recommendation.Description
+                    recommendation.ShortDescription
```
Comment (normj) [inline src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs:124]: Shouldn't we return back both ShortDescription and Description to the toolkit? I through the full Description was going to show up in the detail window or something.

File: src/AWS.Deploy.Recipes/RecipeDefinitions/ASP.NETAppECSFargate.recipe [hunk 2]
```diff
@@ -8,6 +8,7 @@
     "CdkProjectTemplate": "../CdkTemplates/AspNetAppEcsFargate",
     "CdkProjectTemplateId": "netdeploy.AspNetAppEcsFargate",
     "Description": "ASP.NET Core applications built as a container and deployed to Amazon Elastic Container Service (Amazon ECS) with compute power managed by AWS Fargate compute engine. Recommended for applications that can be deployed as a container image. If your project does not contain a Dockerfile, one will be generated for the project.",
+    "ShortDescription": "ASP.NET Core applications built as a container and deployed to Amazon Elastic Container Service (Amazon ECS).",
```
Comment (normj) [inline src/AWS.Deploy.Recipes/RecipeDefinitions/ASP.NETAppECSFargate.recipe:11]: Suggested text:

"ASP.NET Core deployed to Amazon Elastic Container Service (Amazon ECS)."

File: src/AWS.Deploy.Recipes/RecipeDefinitions/ASP.NETAppElasticBeanstalk.recipe [hunk 3]
```diff
@@ -8,6 +8,7 @@
     "CdkProjectTemplate": "../CdkTemplates/AspNetAppElasticBeanstalkLinux",
     "CdkProjectTemplateId": "netdeploy.AspNetAppElasticBeanstalkLinux",
     "Description": "Deploy an ASP.NET Core application to AWS Elastic Beanstalk. Recommended for applications that are not set up to be deployed as containers.",
+    "ShortDescription": "Deploy an ASP.NET Core application to AWS Elastic Beanstalk.",
```
Comment (normj) [inline src/AWS.Deploy.Recipes/RecipeDefinitions/ASP.NETAppElasticBeanstalk.recipe:11]: Suggested text:

"ASP.NET Core application deployed to AWS Elastic Beanstalk on Linux."

File: src/AWS.Deploy.Recipes/RecipeDefinitions/BlazorWasm.recipe [hunk 4]
```diff
@@ -8,6 +8,7 @@
     "CdkProjectTemplate": "../CdkTemplates/BlazorWasm",
     "CdkProjectTemplateId": "netdeploy.BlazorWasm",
     "Description": "Deploy a Blazor WebAssembly application to an Amazon S3 bucket. The Amazon S3 bucket created during deployment will be configured for web hosting and its contents will be open to the public with read access.",
+    "ShortDescription": "Deploy a Blazor WebAssembly application to an Amazon S3 bucket.",
```
Comment (normj) [inline src/AWS.Deploy.Recipes/RecipeDefinitions/BlazorWasm.recipe:11]: Suggested text:

"Blazor WebAssembly application hosted in an Amazon S3 bucket."

File: src/AWS.Deploy.Recipes/RecipeDefinitions/ConsoleAppECSFargateScheduleTask.recipe [hunk 5]
```diff
@@ -8,6 +8,7 @@
     "CdkProjectTemplate": "../CdkTemplates/ConsoleAppECSFargateScheduleTask",
     "CdkProjectTemplateId": "netdeploy.ConsoleAppECSFargateScheduleTask",
     "Description": "Deploy a console application as a scheduled task to Amazon Elastic Container Service (Amazon ECS). Recommended for applications that can be deployed as a container image. If your project does not contain a Dockerfile, one will be generated for the project.",
+    "ShortDescription": "Deploy a console application as a scheduled task to Amazon Elastic Container Service (Amazon ECS).",
```
Comment (normj) [inline src/AWS.Deploy.Recipes/RecipeDefinitions/ConsoleAppECSFargateScheduleTask.recipe:11]: Suggested Text:

"Console application deployed as a scheduled task to Amazon Elastic Container Service (Amazon ECS)."

File: src/AWS.Deploy.Recipes/RecipeDefinitions/ConsoleAppECSFargateService.recipe [hunk 6]
```diff
@@ -8,6 +8,7 @@
     "CdkProjectTemplate": "../CdkTemplates/ConsoleAppECSFargateService",
     "CdkProjectTemplateId": "netdeploy.ConsoleAppEcsFargateService",
     "Description": "Deploy a console application as a service to Amazon Elastic Container Service (Amazon ECS). Recommended for applications that can be deployed as a container image. If your project does not contain a Dockerfile, one will be generated for the project.",
+    "ShortDescription": "Deploy a console application as a service to Amazon Elastic Container Service (Amazon ECS).",
```
Comment (normj) [inline src/AWS.Deploy.Recipes/RecipeDefinitions/ConsoleAppECSFargateService.recipe:11]: Suggested Text:

"Console application deployed as a service to Amazon Elastic Container Service (Amazon ECS)."

**Review by normj — CHANGES_REQUESTED** [review]
My guidance for descriptions is starting with what we think the project is. I don't want them to see ever description starting with the word "Deploy".

File: src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs (hunk 1 above)
Comment (philasmar) [reply src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs:124]: We are doing that, but as far as the RecommendationSummary is concerned, it only cares about the short description. The long description is retrieved via the GetRecipe API

File: src/AWS.Deploy.Recipes/RecipeDefinitions/ASP.NETAppECSFargate.recipe (hunk 2 above)
Comment (philasmar) [reply src/AWS.Deploy.Recipes/RecipeDefinitions/ASP.NETAppECSFargate.recipe]: Fixed

File: src/AWS.Deploy.Recipes/RecipeDefinitions/ASP.NETAppElasticBeanstalk.recipe (hunk 3 above)
Comment (philasmar) [reply src/AWS.Deploy.Recipes/RecipeDefinitions/ASP.NETAppElasticBeanstalk.recipe]: Fixed

File: src/AWS.Deploy.Recipes/RecipeDefinitions/BlazorWasm.recipe (hunk 4 above)
Comment (philasmar) [reply src/AWS.Deploy.Recipes/RecipeDefinitions/BlazorWasm.recipe]: Fixed

File: src/AWS.Deploy.Recipes/RecipeDefinitions/ConsoleAppECSFargateScheduleTask.recipe (hunk 5 above)
Comment (philasmar) [reply src/AWS.Deploy.Recipes/RecipeDefinitions/ConsoleAppECSFargateScheduleTask.recipe]: Fixed

File: src/AWS.Deploy.Recipes/RecipeDefinitions/ConsoleAppECSFargateService.recipe (hunk 6 above)
Comment (philasmar) [reply src/AWS.Deploy.Recipes/RecipeDefinitions/ConsoleAppECSFargateService.recipe]: Fixed

**Comment by ganeshnj** [comment]
What happened to the idea to truncation when (full) description is longer the displaying UI element?

**Comment by philasmar** [comment]
> What happened to the idea to truncate when (full) description is longer the displaying UI element?

We had a meeting with the UX/UI expert and decided that we will add a new field.

**Review by ganeshnj — APPROVED** [review]

**Review by normj — APPROVED** [review]

**Pull request merged**